### ISSUE TITLE
Get.oturep error with v.1.41.3

### ISSUE THREAD

**Issue by Esmee73** [mothur/mothur#590]
Hi, 

I get an error message when running the following get.oturep command in v.1.41.3, while it worked in  v.1.39.5, the previous version that I used. 

It only works in v.1.41.3 when I set label=0. It doesn't work for other labels or for multiple labels.

mothur > get.oturep(column=final.pick.Cyanobacteria.dist, name=final.pick.Cyanobacteria.names, list=final.pick.Cyanobacteria.opti_mcc.list, fasta=final.pick.Cyanobacteria.fasta, label=0-0.01-0.02-0.03-0.04-0.05-0.06, group=final.pick.Cyanobacteria.groups)
/******************************************/
/nRunning command: count.seqs(name=final.pick.Cyanobacteria.names, group=final.pick.Cyanobacteria.groups)

It took 0 secs to create a table for 45375 sequences.

Total number of sequences: 45375

Output File Names:
final.pick.Cyanobacteria.count_table

/nRunning command: get.seqs(list=final.pick.Cyanobacteria.opti_mcc.list, accnos=temp.accnos)
Selected 36 sequences from your list file.

Output File Names:
final.pick.Cyanobacteria.opti_mcc.0.pick.list
final.pick.Cyanobacteria.opti_mcc.0.01.pick.list
final.pick.Cyanobacteria.opti_mcc.0.02.pick.list
final.pick.Cyanobacteria.opti_mcc.0.03.pick.list
final.pick.Cyanobacteria.opti_mcc.0.04.pick.list
final.pick.Cyanobacteria.opti_mcc.0.05.pick.list
final.pick.Cyanobacteria.opti_mcc.0.06.pick.list

/******************************************/
[ERROR]: std::bad_allocRAM used: 0.0162811Gigabytes . Total Ram: 31.3648Gigabytes.

has occurred in the GetOTURepCommand class function execute. This error indicates your computer is running out of memory.  This is most commonly caused by trying to process a dataset too large, using multiple processors, or a file format issue. If you are using multiple processors, try running the command with processors=1, the more processors you use the more memory is required. Also, you may be able to reduce the size of your dataset by using the commands outlined in the Schloss SOP, http://www.mothur.org/wiki/Schloss_SOP. If you are unable to resolve the issue, please contact Pat Schloss at mothur.bugs@gmail.com, and be sure to include the mothur.logFile with your inquiry.

Do you know how this may come?

Many thanks in advance,
Esmee

**Comment by mothur-westcott** [comment]
Thanks for reporting this issue. The name / group option is not working as expected. Mothur generates a count file from the name and group file to save memory during processing. Mothur then uses the get.seqs command to create a list file for each distance that contains only the unique names. The issue occurs because mothur is only looking at the first list file label it gets from the get.seqs parameter.  I will fix this for our next release. In the meantime, you can work around the issue by running the get.oturep command multiple times using the count parameter and a list file for the distance you want to find reps for.

mothur > count.seqs(name=final.pick.Cyanobacteria.names, group=final.pick.Cyanobacteria.groups) - create count table from your name and group file.
mothur > list.seqs(count=current) - list names in count file
mothur > get.seqs(list=final.pick.Cyanobacteria.opti_mcc.list, accnos=current) - select unique names from list file, will create a list file for each label
mothur > get.oturep(column=final.pick.Cyanobacteria.dist, count=current, list=final.pick.Cyanobacteria.opti_mcc.0.01.pick.list, fasta=final.pick.Cyanobacteria.fasta) - generate otureps for label 0.01
mothur > get.oturep(column=final.pick.Cyanobacteria.dist, count=current, list=final.pick.Cyanobacteria.opti_mcc.0.03.pick.list, fasta=final.pick.Cyanobacteria.fasta) - generate otureps for label 0.03
...

**Comment by Esmee73** [comment]
Thank you very much for your reply! Thank you for providing a workaround and I will keep an eye on the new release!

**Comment by Esmee73** [comment]
Hi, 

I implemented your workaround suggestion. I get the following error message for label equal to or greater than 0.02: "Sequence 'xxx' was not found in the name or count file, please correct". 

I read in previous posts that this type of error is usually caused by a file mismatch. May it be that count=current always uses a count table that is one label smaller than the list file?

i.e. count=current uses final.pick.Cyanobacteria.count_table for list=final.pick.Cyanobacteria.opti_mcc.0.pick.list, count=final.pick.Cyanobacteria.opti_mcc.0.pick.0.rep.count_table for list=final.pick.Cyanobacteria.opti_mcc.0.01.pick.list, count=final.pick.Cyanobacteria.opti_mcc.0.01.pick.0.01.rep.count_table for list=final.pick.Cyanobacteria.opti_mcc.0.02.pick.list, etc.

Would it be ok to use count=final.pick.Cyanobacteria.count_table (instead of count=current) to generate otureps for all labels? 

Many thanks in advance,
Esmee

**Comment by Esmee73** [comment]
Hi, 

I've tried to implement the workaround you suggested. I however get an error message for the get.oturep command for label equal to or greater than 0.02 (and I am not sure whether it works correctly for label 0.01 and 0 either). 

The error message is: "Sequence 'xxx' was not found in the name or count file, please correct". 

I saw for other issues posted on GitHub that this error is mostly caused by a file mismatch. 

I think the count table and list file do not match, because they do not have the same label, i.e. when count=current it uses 
final.pick.Cyanobacteria.count_table when list=final.pick.Cyanobacteria.opti_mcc.0.pick list, final.pick.Cyanobacteria.opti_mcc.0.pick.0.rep.count_table when list= final.pick.Cyanobacteria.opti_mcc.0.01.pick.list, final.pick.Cyanobacteria.opti_mcc.0.01.pick.0.01.rep.count_table when list=final.pick.Cyanobacteria.opti_mcc.0.02.pick.list, etc.

I am not sure whether this causes the problem, but I've tried to workaround this by creating label specific count tables using make.shared command - that I can then use as count tables for get.oturep. 
With make.shared I however get OTU labels instead of representative sequences in count tables, so this does not work either. 

In short, I don't know how I can get get.oturep to work.
Could you please tell me how I can overcome this problem?

Many thanks in advance,
Esmee

**Comment by mothur-westcott** [comment]
You are correct, I shouldn't have set count to correct.  Have you tried this?

mothur > count.seqs(name=final.pick.Cyanobacteria.names, group=final.pick.Cyanobacteria.groups) - create count table from your name and group file.
mothur > list.seqs(count=current) - list names in count file
mothur > get.seqs(list=final.pick.Cyanobacteria.opti_mcc.list, accnos=current) - select unique names from list file, will create a list file for each label
mothur > get.oturep(column=final.pick.Cyanobacteria.dist, count=final.pick.Cyanobacteria.count_table, list=final.pick.Cyanobacteria.opti_mcc.0.01.pick.list, fasta=final.pick.Cyanobacteria.fasta) - generate otureps for label 0.01
mothur > get.oturep(column=final.pick.Cyanobacteria.dist, count=final.pick.Cyanobacteria.count_table, list=final.pick.Cyanobacteria.opti_mcc.0.03.pick.list, fasta=final.pick.Cyanobacteria.fasta) - generate otureps for label 0.03

**Comment by Esmee73** [comment]
I had tried this and it run, but I was not sure whether it was correct. But then I will do it like this! Many thanks for your help!

**Comment by mothur-westcott** [comment]
Glad to help! Thanks for reporting this issue!
